src/Fixer/ControlStructure/SingleExpressionPerLineFixer.php

```diff
@@ -0,0 +1,357 @@
+        $fixControlStructures = \in_array(self::ELEMENTS_CONTROL_STRUCTURES, $this->configuration['elements'], true);
+        $fixSwitchCases = \in_array(self::SWITCH_CASES, $this->configuration['elements'], true);
+        $fixMatch = \in_array(self::MATCH_EXPRESSIONS, $this->configuration['elements'], true);
+
+        for ($index = $begin + 1; $index < $end + $tokenAddedCount; ++$index) {
+            /** @var Token $token */
+            $token = $tokens[$index];
+
+            if (
+                null !== $lineDelimiters
+                && $token->equalsAny($lineDelimiters)
+                && (null === $lineDelimitersIndexes || \in_array($index, $lineDelimitersIndexes, true))
+            ) {
+                if ($shouldAddNewLine) {
+                    $tokenAddedCount += $this->addNewLineAfterIfNecessary($tokens, $index);
+                }
+
+                continue;
+            }
+
+            if ($tokens[$index]->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $index);
+                $added = $this->processBlock($tokens, $index, $until, $fixArrays, [',']);
+
+                $index = $until + $added;
+                $tokenAddedCount += $added;
+
+                continue;
+            }
+
+            if (!$tokens[$index]->equals('(')) {
+                continue;
+            }
+
+            $prevIndex = $tokens->getPrevMeaningfulToken($index);
+            if ($tokens[$prevIndex]->isGivenKind(T_ARRAY)) {
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+                $added = $this->processBlock($tokens, $index, $until, $fixArrays, [',']);
+
+                $index = $until + $added;
+                $tokenAddedCount += $added;
+
+                continue;
+            }
+
+            if ($tokens[$prevIndex]->isGivenKind([T_IF, T_ELSEIF, T_WHILE, T_FOREACH, T_CATCH])) {
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+                $added = $this->processBlock($tokens, $index, $until, $fixControlStructures);
+
+                $index = $until + $added;
+                $tokenAddedCount += $added;
+
+                continue;
+            }
+
+            if ($tokens[$prevIndex]->isGivenKind(T_FOR)) {
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+                $added = $this->processBlock($tokens, $index, $until, $fixControlStructures, [';']);
+
+                $index = $until + $added;
+                $tokenAddedCount += $added;
+
+                continue;
+            }
+
+            $prevPrevIndex = $tokens->getPrevMeaningfulToken($prevIndex);
+            if ($tokens[$prevIndex]->equalsAny([']', [T_CLASS], [T_STRING], [T_VARIABLE], [T_STATIC]])
+                && !$tokens[$prevPrevIndex]->isGivenKind(T_FUNCTION)
+            ) {
```

Comment:
This should have been fixed automatically 😅
```suggestion
            if (
                $tokens[$prevIndex]->equalsAny([']', [T_CLASS], [T_STRING], [T_VARIABLE], [T_STATIC]])
                && !$tokens[$prevPrevIndex]->isGivenKind(T_FUNCTION)
            ) {
```

Reply:
PHPCSFixer extends Symfony ruleset which disable the FullyMultilineFixer...because it will do too much change to the symfony codebase.

Reply:
If the rule is wanted by Symfony but temporarily disabled for practical reasons, I'd say it should be enabled in `@Symfony` ruleset (so it can be applied to e.g. third party bundles) and disabled in the Symfony repository.

Reply:
Hi @nicolas-grekas.
Sorry to take your time.

Do you think this new rule `single_expression_per_line` will be desired by Symfony ?
It's made to respect the PSR12
- [In control structures, ]expressions in parentheses MAY be split across multiple lines, where each subsequent line is indented at least once. When doing so, the first condition MUST be on the next line. 
- Argument lists and variable lists MAY be split across multiple lines, where each subsequent line is indented once. When doing so, the first item in the list MUST be on the next line, and there MUST be only one argument or variable per line.

And PERCS 2
- Same for arrays

So it will fix
```
if ($a
    || $b) {
}

$arrayUntouched  = [1, 2, 3];
$array = [1, 2, [
    $a,
    $b,
]];

$resultUntouched  = $this->foo($a, [$b, $c]);
$result = $this->foo($a, [
    $b, 
    $c,
]);
```
to
```
if (
    $a
    || $b
) {
}

$arrayUntouched  = [1, 2, 3];
$array = [
    1,
    2,
    [
        $a,
        $b,
    ],
];

$resultUntouched  = $this->foo($a, [$b, $c]);
$result = $this->foo(
    $a,
    [
        $b, 
        $c,
    ],
);
```